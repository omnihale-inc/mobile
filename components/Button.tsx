import { ReactNode } from "react";
import { StyleSheet, Pressable } from "react-native";

import { Text } from "./Themed";

export const GreenButton = ({
  children,
  onPress
}: {
  children: ReactNode | ReactNode[];
  onPress: () => void;
}) => {
  return (
    <Pressable onPress={onPress}>
      <Text style={styles.GreenButton}>{children}</Text>
    </Pressable>
  );
};

export const GreyButton = ({
  children
}: {
  children: ReactNode | ReactNode[];
}) => {
  return <Text style={styles.GreyButton}>{children}</Text>;
};

const styles = StyleSheet.create({
  GreenButton: {
    fontFamily: "SatoshiMedium",
    color: "#ffffff",
    backgroundColor: "#237e5d",
    textAlign: "center",
    paddingVertical: 22,
    fontSize: 14,
    borderRadius: 30,
    marginHorizontal: 24,
    marginBottom: 40
  },
  GreyButton: {
    fontFamily: "SatoshiMedium",
    color: "#000000",
    backgroundColor: "#d1d3d9",
    textAlign: "center",
    paddingVertical: 22,
    fontSize: 14,
    borderRadius: 30,
    marginHorizontal: 24,
    marginBottom: 40
  }
});
