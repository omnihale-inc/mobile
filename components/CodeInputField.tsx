import React from "react";
import {
  NativeSyntheticEvent,
  StyleProp,
  TextInput,
  TextInputKeyPressEventData,
  TextStyle
} from "react-native";

export type CodeInputFieldProps = {
  events: {
    onChangeTextHandler(changeValue: string): void;
    onKeyPressHandler(
      key: NativeSyntheticEvent<TextInputKeyPressEventData>
    ): void;
    onFocusHandler(): void;
    onBlurHandler(): void;
  };
  value: string;
  style: StyleProp<TextStyle>;
};

const CodeInputField = React.forwardRef(
  (
    { events, value, style }: CodeInputFieldProps,
    ref: React.ForwardedRef<TextInput | null>
  ) => {
    return (
      <TextInput
        onChangeText={(changeValue) => events.onChangeTextHandler(changeValue)}
        value={value}
        style={style}
        onFocus={events.onFocusHandler}
        onBlur={events.onBlurHandler}
        keyboardType="numeric"
        ref={ref}
        onKeyPress={(key) => events.onKeyPressHandler(key)}
      />
    );
  }
);

export default CodeInputField;
