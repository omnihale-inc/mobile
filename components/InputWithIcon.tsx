import { ReactNode } from "react";
import { StyleSheet, TextInput } from "react-native";

import { View } from "./Themed";

const InputWithIcon = ({
  value,
  onChangeText,
  icon
}: {
  value?: string;
  onChangeText?: (value: string) => void;
  icon: (props?: object) => ReactNode;
}) => {
  return (
    <View style={styles.emailInputWrapper}>
      <TextInput
        style={styles.emailInput}
        placeholder="Enter Email address"
        keyboardType="email-address"
        value={value}
        onChangeText={onChangeText}
      />
      <View style={styles.emailIcon}>{icon()}</View>
    </View>
  );
};

const styles = StyleSheet.create({
  emailInputWrapper: {
    position: "relative"
  },
  emailInput: {
    borderColor: "#d1d3d9",
    fontFamily: "SatoshiRegular",
    borderWidth: 1,
    marginVertical: 24,
    paddingVertical: 15,
    paddingLeft: 55,
    paddingRight: 40,
    borderRadius: 30,
    fontSize: 17
  },
  emailIcon: { position: "absolute", top: 43, left: 22 }
});

export default InputWithIcon;
