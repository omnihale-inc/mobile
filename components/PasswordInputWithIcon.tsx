import { ReactNode, useState } from "react";
import { StyleSheet, TextInput } from "react-native";

import { View } from "./Themed";
import EyeInvisibleIcon from "./icons/EyeInvisible";
import EyeVisibleIcon from "./icons/EyeVisible";

const PasswordInputWithIcon = ({
  value,
  onChangeText,
  icon,
  placeholder
}: {
  value?: string;
  onChangeText?: (value: string) => void;
  icon: (props?: object) => ReactNode;
  secureText?: boolean;
  placeholder?: string;
}) => {
  const [passwdIsDisplayed, setPasswdIsDisplayed] = useState(false);

  return (
    <View>
      <View style={styles.emailInputWrapper}>
        <TextInput
          style={styles.emailInput}
          placeholder={placeholder}
          keyboardType="default"
          secureTextEntry={!passwdIsDisplayed}
          value={value}
          onChangeText={onChangeText}
        />
        <View style={styles.emailIcon}>{icon()}</View>
      </View>
      <View
        style={{
          position: "absolute",
          top: 25,
          right: 12,
          backgroundColor: "transparent"
        }}
      >
        {passwdIsDisplayed ? (
          <EyeVisibleIcon
            onPress={() => setPasswdIsDisplayed(!passwdIsDisplayed)}
          />
        ) : (
          <EyeInvisibleIcon
            onPress={() => setPasswdIsDisplayed(!passwdIsDisplayed)}
          />
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  emailInputWrapper: {
    position: "relative"
  },
  emailInput: {
    borderColor: "#d1d3d9",
    fontFamily: "SatoshiRegular",
    borderWidth: 1,
    marginVertical: 24,
    paddingVertical: 15,
    paddingLeft: 55,
    paddingRight: 40,
    borderRadius: 30,
    fontSize: 17
  },
  emailIcon: { position: "absolute", top: 43, left: 22 }
});

export default PasswordInputWithIcon;
