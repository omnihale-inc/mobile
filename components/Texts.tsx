import { ReactNode } from "react";
import { TextStyle } from "react-native";

import { Text } from "./Themed";

export const BoldText = ({
  style,
  children
}: {
  children: ReactNode;
  style?: TextStyle;
}) => <Text style={{ fontFamily: "SatoshiBold", ...style }}>{children}</Text>;

export const RegularText = ({
  children,
  style
}: {
  children: ReactNode;
  style?: TextStyle;
}) => (
  <Text style={{ fontFamily: "SatoshiRegular", ...style }}>{children}</Text>
);

export const MediumText = ({
  children,
  style
}: {
  children: ReactNode;
  style?: TextStyle;
}) => <Text style={{ fontFamily: "SatoshiMedium", ...style }}>{children}</Text>;
