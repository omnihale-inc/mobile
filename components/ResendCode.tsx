import { ReactNode } from "react";
import { Pressable, TextStyle, View, ViewStyle } from "react-native";

import { RegularText } from "./Texts";

const ResendCode = ({
  resend,
  onPress,
  styles,
  countDown
}: {
  resend: boolean;
  onPress: () => void;
  styles: { resend: ViewStyle; greenText: TextStyle };
  countDown: () => ReactNode;
}) => {
  return !resend ? (
    <View style={styles.resend}>
      <RegularText>Resend code in</RegularText>
      {countDown()}
    </View>
  ) : (
    <Pressable onPress={onPress}>
      <RegularText style={{ paddingTop: 12, ...styles.greenText }}>
        Resend code
      </RegularText>
    </Pressable>
  );
};

export default ResendCode;
