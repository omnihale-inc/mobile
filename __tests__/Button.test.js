import renderer from "react-test-renderer";

import { GreyButton, GreenButton } from "../components/Button";

it("render grey button", () => {
  const tree = renderer.create(<GreyButton>grey button</GreyButton>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("render green button", () => {
  const tree = renderer
    .create(<GreenButton onPress={jest.fn()}>green button</GreenButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
