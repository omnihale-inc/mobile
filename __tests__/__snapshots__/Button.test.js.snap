// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render green button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "backgroundColor": "#237e5d",
          "borderRadius": 30,
          "color": "#ffffff",
          "fontFamily": "SatoshiMedium",
          "fontSize": 16,
          "marginBottom": 40,
          "marginHorizontal": 24,
          "paddingVertical": 22,
          "textAlign": "center",
        },
      ]
    }
  >
    green button
  </Text>
</View>
`;

exports[`render grey button 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      {
        "backgroundColor": "#d1d3d9",
        "borderRadius": 30,
        "color": "#000000",
        "fontFamily": "SatoshiMedium",
        "fontSize": 16,
        "marginBottom": 40,
        "marginHorizontal": 24,
        "paddingVertical": 22,
        "textAlign": "center",
      },
    ]
  }
>
  grey button
</Text>
`;
